src/include/ipc/topology.h

```diff
@@ -287,4 +287,9 @@ struct sof_ipc_pipe_comp_connect {
 	uint32_t sink_id;
 } __attribute__((packed));
 
+/* extended data struct for component new */
+struct sof_ipc_comp_new_ext {
+	uint8_t uuid[16];
```

Comment:
can you add a reserved field here too.

Reply:
Will need a macro for UUID size insread of 16.

Reply:
> can you add a reserved field here too.

As explained to @ktrzcinx here https://github.com/thesofproject/linux/pull/2090#discussion_r441301317, I don't think we need reserved field here.

Reply:
> Will need a macro for UUID size insread of 16.

We already have 'UUID_SIZE' in both FW and driver, but as they are defined in the specific uuid.h, to avoid to include uuid header file to topology.h, are you suggesting to add another one here in this topology.h which will be shared between FW and driver?

Reply:
I mean define the size in ABI header which will be included by internal FW/driver header that defines the UUID structure/APIs.

Reply:
the UUID structure/APIs is a independent code and should not have the dependency to the ABI header (at least at driver side it is like this, the UUID is not only used by audio), there are already 'UUID_SIZE' definition there. if we add new macro, it will be used in the ABI and IPC struct only.